pkg/sql/exec_util.go

```diff
@@ -1057,6 +1057,17 @@ func golangFillQueryArguments(args ...interface{}) (tree.Datums, error) {
 				switch {
 				case val.IsNil():
 					d = tree.DNull
+				case val.Type().Elem().Kind() == reflect.String:
+					a := tree.NewDArray(types.String)
+					// TODO(angelaw): Ranging through val seems to require pointer arithmetic so did it this way
+					for s := range args {
+						// It is a bit hacky though...
+						if err := a.Append(tree.NewDString(reflect.ValueOf(s).String())); err != nil {
+							return nil, err
+						}
+					}
+					res[i] = a
```

Comment:
no need to set res[i] here -- that can happen on line 1073 if we just make the variable `d = tree.NewDArray(types.String)`

also i think it would be a bug to return early here, since it has to go through the rest of the outer for loop

Reply:
I originally had done that, but `d` is a `tree.Datum` and doesn't have an `Append` method so I thought I'd make it a separate method. I see what you mean about returning too early though.

Reply:
ooh got it. then i guess instead of `res[i] = a` this line should just be `d = a`